hack/generator/pkg/astmodel/errored_type.go

```diff
@@ -20,20 +20,24 @@ type ErroredType struct {
 	warnings []string
 }
 
-func MakeErroredType(t Type, errors []string, warnings []string) ErroredType {
-	return ErroredType{
+var _ Type = &ErroredType{}
+
+func MakeErroredType(t Type, errors []string, warnings []string) *ErroredType {
```

Comment:
minor: `NewErroredType` if it returns a ptr is I think the style we've been using